src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileProvider.cs

```diff
@@ -18,6 +18,8 @@ public class PhysicalFileProvider : IFileProvider, IDisposable
         private const string PollingEnvironmentKey = "DOTNET_USE_POLLING_FILE_WATCHER";
         private static readonly char[] _invalidFileNameChars = Path.GetInvalidFileNameChars()
             .Where(c => c != Path.DirectorySeparatorChar && c != Path.AltDirectorySeparatorChar).ToArray();
+        private static readonly char[] _invalidFilterChars = _invalidFileNameChars
```

Comment:
Found an issue when I was adding tests that invalid characters were not being checked for the provider watcher token. Since handling of leading slashes affect those filters, I added the check to make the behaviour consistent across files, directories, and watcher.